packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageModule.kt

```diff
@@ -39,6 +50,65 @@ class ExpoImageModule(val context: ReactApplicationContext) : ReactContextBaseJa
     }
   }
 
+  @ReactMethod
+  fun queryCache(urls: ReadableArray, promise: Promise) {
+    moduleCoroutineScope.launch(
+        CoroutineExceptionHandler { _, e ->
+          promise.reject("ERR_IMAGE_QUERY_CACHE_FAILURE", "Failed to read the cache: ${e.message}", e)
+        }
+    ) {
+      val resultMap = WritableNativeMap()
+      urls.toArrayList()
+        .filterIsInstance<String>()
+        .map { url ->
+          async {
+            val isInDiskCache = isInCache(url, ImageCacheType.DISK)
+            val isInMemoryCache = isInCache(url, ImageCacheType.MEMORY)
+            Triple(url, isInDiskCache, isInMemoryCache)
+          }
+        }
+        .awaitAll()
+        .filter {
+          (_, inDisk, inMemory) -> inDisk || inMemory
+        }
+        .forEach{ (url, inDisk, inMemory) ->
+          resultMap.putString(
+            url,
+            if (inDisk && inMemory) {
+              "disk/memory"
+            } else if (inDisk) {
+              "disk"
+            } else {
+              "memory"
+            }
+          )
```

Comment:
Just a side note - I just discovered it can be done this way 😱 had no idea that `when` can be used without specifying a single subject.
```suggestion
          val type = when {
            inDisk && inMemory -> "disk/memory"
            inDisk -> "disk"
            else -> "memory"
          }
          resultMap.putString(url, type)
```